public/pages/CreateMonitor/components/ClusterMetricsMonitor/ClusterMetricsMonitor.js

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState } from 'react';
+import _ from 'lodash';
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiCodeEditor,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiLink,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiOverlayMask,
+  EuiSpacer,
+  EuiText,
+} from '@elastic/eui';
+import { isInvalidApiPath } from '../../../../utils/validate';
+import { FormikComboBox, FormikFieldText } from '../../../../components/FormControls';
+import {
+  API_PATH_REQUIRED_PLACEHOLDER_TEXT,
+  EMPTY_PATH_PARAMS_TEXT,
+  REST_API_REFERENCE,
+  API_TYPES,
+} from './utils/clusterMetricsMonitorConstants';
+import {
+  getApiPath,
+  getExamplePathParams,
+  isInvalidApiPathParameter,
+  validateApiPathParameter,
+} from './utils/clusterMetricsMonitorHelpers';
+import { FORMIK_INITIAL_VALUES } from '../../containers/CreateMonitor/utils/constants';
+
+const setApiType = (form, apiType) => {
+  const pathParams = _.get(form, 'uri.path_params', FORMIK_INITIAL_VALUES.uri.path_params);
+  form.setFieldValue('uri.api_type', apiType);
+  form.setFieldValue('uri.path', getApiPath(_.isEmpty(pathParams), apiType));
+};
+
+const renderModal = (closeModal, prevApiType, selectedApiType, form) => {
+  const onClear = () => {
+    setApiType(form, selectedApiType);
+    form.setFieldValue('triggerDefinitions', []);
+    closeModal();
+  };
+  const onKeep = () => {
+    setApiType(form, selectedApiType);
+    closeModal();
+  };
+  const onClose = () => {
+    setApiType(form, prevApiType);
+    closeModal();
+  };
+  return (
+    <EuiOverlayMask>
+      <EuiModal onClose={onClose} data-test-subj={'clusterMetricsClearTriggersModal'}>
```

Comment:
Also we can consider extracting this modal to a separate class. Would be easier to write a basic jest test and will be reusable. (Not urgent to fix)

Reply:
Agreed. For now, I'll leave this implementation as-is, but we can extract and make the modal more generic to address this issue I create https://github.com/opensearch-project/alerting-dashboards-plugin/issues/188